packages/virtual-core/src/index.ts

```diff
@@ -330,15 +351,17 @@ export class Virtualizer<
     const get = () => {
       if (_ro) {
         return _ro
-      } else if (typeof ResizeObserver !== 'undefined') {
-        return (_ro = new ResizeObserver((entries) => {
-          entries.forEach((entry) => {
-            this._measureElement(entry.target as TItemElement, entry)
-          })
-        }))
-      } else {
+      }
+
+      if (typeof ResizeObserver === 'undefined' || !this.targetWindow) {
```

Comment:
I think here we would like to check if ResizeObserver is on targetWindow?

Reply:
Do you mean a check like:
```tsx
      if (
        typeof ResizeObserver === 'undefined' ||
        !this.targetWindow ||
        !this.targetWindow.ResizeObserver
      ) {
        return null
      }
```
I think it's not necessary because this should be guarded by type. When used in SSR `this.targetWindows` (`this.scrollElement.ownerDocument.defaultView`) should be null

Reply:
No, imho we don't need to check ResizeObserver on global context if we read it from this.targetWindow

```tsx
if (!this.targetWindow || !this.targetWindow.ResizeObserver) {
  return null
}
```

Reply:
Right! I updated it according to the suggestion

Reply:
Thanks.